### DISCUSSION TITLE
Any plans to make an Android & IOS version?

### DISCUSSION THREAD

**Discussion by rinka-meltiron** [zim-desktop-wiki/zim-desktop-wiki#1737] [Ideas]
I was just talking to a friend about Zim and mentioned I found it very effective.  He said he uses his phone extensively (his primary interface) and I though it would be very useful to have a release on Android / IOS.

Yes I know that means two more OSes but the mobile platform is becoming ever popular and the ability to access my Zim Notebooks on Android would be extremely useful.

If something like that happened, then the ability to have my notebook stored on (or at least replicated to) Google Drive or Box or Dropbox would also have great value.

**Comment by jaap-karssenberg** [comment]
No. The toolkits used to build zim cannot just be deployed on a phone OS,
so basically to build an app would mean a whole new development.

My recommendation is to use the "Markor" app instead. I does support zim
wiki syntax in case you want to do a shared notebook.

My mid- to long-term plan is to make Zim support Markdown syntax, which is
supported by many apps.

-- Jaap

**Reply by introt** [reply]
To expand a bit on this:

> If something like that happened, then the ability to have my notebook stored on (or at least replicated to) Google Drive or Box or Dropbox would also have great value.

This is already possible; you just need to store your notebook inside a synced folder. I'm personally using Markor + Syncthing to have my notes on the go.

**Reply by rinka-meltiron** [reply]
Thank you so much for your responses.  I'll explore Markor for myself.

One suggestion: If you are moving to Markdown, I'm sure you will be considering a wysiwyg front end.  Do consider having a stand-alone front end (or perhaps a plug-in for libreoffice/MS Word) that emits .md files.  We don't have good wsyiwyg markdown editors today.

BTW, I just saw zim-android-mockapp - this just might be the answer to what I was looking for.

**Comment by Nzalo** [comment]
Hello everyone,

I found this question looking for a way to read my Zim wiki on iOS devices.
This might be of help for anyone going through the same rabbit hole.

After testing Koder, Runestone, iWriterPro(paid), Pretext and Textastic (paid), the only app with some syntax highlighting for the .txt files was Textastic (DokuWiki and Mediawiki). None of them could preview the pages as we are used to.

Navigating through the file structure is a bit cumbersome and I still haven't found a solution. If so, I will post an update here.

**Reply by Nzalo** [reply]
The native Files app did it. 
Example to use the journal functionality: sort files as list + group by type + order descendently
Set a bookmark (favorite) for the journal folders of the wikis you are connected to. 
Open .txt by default with Textastic.

**Comment by crazy2be** [comment]
For what it's worth, there is an Android port of Gtk in the works. Maybe in a few years we can live the dream of Zim on Android :grinning:. I don't imagine iOS support will come anytime soon, however.

https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/7555

(Of course, the real challenge is sync. I would be quite happy with a simple plain text version of zim on my Android if the sync worked reliably and I was not worried about losing/overwriting data.)

**Comment by vince06fr** [comment]
On Android, you can use termux with termux-setup-storage and use Git,  Bazaar, Mercurial, rsync, or whatever for sync and using markor with the synced file
https://termux.dev/en/
https://wiki.termux.com/wiki/Termux-setup-storage

Maybe we can port zim for android and ios with kivy 
https://kivy.org/

**Comment by kaska** [comment]
Is there any possibility to read zim notes  **with pictures** on android? I checked markor but:
1. it looks like clear txt file - no pictures
2. totally no structure just simple directories and txt files
Is there any possibility to read it with files in any familiar way? Is there any step which I'm missing? I can live without synchro, I jast need fast simple way to read already created notes with lot of pictures

**Reply by introt** [reply]
If a read-only copy is sufficient, you can export the wiki in eg. html and read that on the Android device. If you don't want to export after each change, you might be able to run Zim's integrated web server in Termux, or serve from your PC over Tailscale or similar.

**Reply by introt** [reply]
I've been toying around with the idea of including either the full page source or just an empty TextInput for appending to the page in the html template to make non-destructive additions from client devices over the network, but haven't had the time to implement any of that so far

**Reply by krisztianmukli** [reply]
Markor can handle images and links perfectly, IF you turn on Reader Mode. By default, it opens files in Editor mode. Check the settings or try the eye-shaped icon in the top icon bar of the editor mode.
[See more info](https://github.com/gsantner/markor)

**Reply by kaska** [reply]
> Markor can handle images and links perfectly, IF you turn on Reader Mode. By default, it opens files in Editor mode. Check the settings or try the eye-shaped icon in the top icon bar of the editor mode. [See more info](https://github.com/gsantner/markor)

@krisztianmukli  thx  I can see that when I  turn on Reader Mode I can see pictures but it's really hard to navigate because I don't see Index and I have complicated structure of notes (lot of directories with notes directly in directory and additional notes in sub-directories where part of them also have sub-directories). When I export it to markdown I can only navigate through directory structure and I don't see any  distinction between directory with additional sub-directories and directory containing only images to notes.

**Comment by Statisticonomicon** [comment]
Hello all,

I just built an app for Android using Claude, yes, vibe coded it like an idiot, except I am not a complete idiot, there are parts missing, it took me a couple of days and it works, so far.

TEST WITH CAUTION and please give me feedback

https://github.com/Statisticonomicon/Zim-Wiki-for-Android-ALPHA
